pkg/provider/helm/values.go

```diff
@@ -4,171 +4,430 @@
+		serverConfig:             serverConfig,
+		serviceType:              "LoadBalancer",
+	}
+	spireServerValues, err := ssv.generateValues()
+	if err != nil {
+		return nil, err
+	}
+
+	spireServer, ok := getNestedMap(spireServerValues, "spire-server")
+	if !ok {
+		return nil, fmt.Errorf("failed to get spire-server map from spireServerValues")
+	}
+
+	controllerManager, ok := getNestedMap(spireServer, "controllerManager")
+	if !ok {
+		return nil, fmt.Errorf("failed to get controllerManager map from spireServer")
+	}
+
+	// Enables the default ClusterSPIFFEID CR by default.
+	controllerManager["identities"] = map[string]any{
+		"clusterSPIFFEIDs": map[string]any{
+			"default": map[string]any{
+				"enabled": true,
+			},
+		},
+	}
+
+	identities, ok := getNestedMap(controllerManager, "identities")
+	if !ok {
+		return nil, fmt.Errorf("failed to get identities map from controllerManager")
+	}
+
 	if len(g.trustZone.AttestationPolicies) > 0 {
-		spireServerValues[`"spire-server"."controllerManager"."identities"."clusterSPIFFEIDs"."default"."enabled"`] = false
+		csids, ok := getNestedMap(identities, "clusterSPIFFEIDs")
+		if !ok {
+			return nil, fmt.Errorf("failed to get clusterSPIFFEIDs map from identities")
+		}
+
+		// Disables the default ClusterSPIFFEID CR.
+		csids["default"] = map[string]any{
+			"enabled": false,
+		}
+
+		// Adds the attestation policies as ClusterSPIFFEID CRs to be reconciled by the spire-controller-manager.
 		for _, binding := range g.trustZone.AttestationPolicies {
 			policy, err := g.source.GetAttestationPolicy(binding.Policy)
 			if err != nil {
 				return nil, err
 			}
+
 			clusterSPIFFEIDs, err := attestationpolicy.NewAttestationPolicy(policy).GetHelmConfig(g.source, binding)
 			if err != nil {
 				return nil, err
 			}
-			spireServerValues[fmt.Sprintf(`"spire-server"."controllerManager"."identities"."clusterSPIFFEIDs"."%s"`, policy.Name)] = clusterSPIFFEIDs
+
+			csids[policy.Name] = clusterSPIFFEIDs
 		}
-	} else {
-		// defaults to true
-		spireServerValues[`"spire-server"."controllerManager"."identities"."clusterSPIFFEIDs"."default"."enabled"`] = true
 	}
 
-	// add federations as clusterFederatedTrustDomains to be reconcilced by spire-controller-manager
+	// Adds the federations as ClusterFederatedTrustDomain CRs to be reconciled by the spire-controller-manager.
 	if len(g.trustZone.Federations) > 0 {
 		for _, fed := range g.trustZone.Federations {
 			tz, err := g.source.GetTrustZone(fed.To)
 			if err != nil {
 				return nil, err
 			}
+
 			if tz.GetBundleEndpointUrl() != "" {
-				spireServerValues[`"spire-server"."federation"."enabled"`] = true
-				config := federation.NewFederation(tz).GetHelmConfig()
-				spireServerValues[fmt.Sprintf(`"spire-server"."controllerManager"."identities"."clusterFederatedTrustDomains"."%s"`, fed.To)] = config
+				spireServer["federation"] = map[string]any{
+					"enabled": true,
+				}
+
+				cftd, ok := getNestedMap(identities, "clusterFederatedTrustDomains")
```

Comment:
This could be tidied up a bit with a getOrCreateNestedMap function:
```suggestion
				cftd, err := getOrCreateNestedMap(identities, "clusterFederatedTrustDomains")
				if err != nil {
				    return nil, err
				}
				cftd[fed.To] = federation.NewFederation(tz).GetHelmConfig()
```